snap-core/src/main/java/org/esa/snap/core/util/ImageUtils.java

```diff
@@ -44,6 +45,77 @@
  */
 public class ImageUtils {
 
+    public static ImageLayout buildMosaicImageLayout(Integer dataBufferType, int imageWidth, int imageHeight, int level) {
```

Comment:
In the ImageUtils class there is another method called 'buildTileImageLayout' invoked when created the tileOpImage object of the final mosaic image.
The two methods (buildMosaicImageLayout and buildTileImageLayout) have different implementation and return the same object type.

Reply:
Yes, I also don't see a need why the method should have Tile in its name.
Actually both methods look very similar from their implemenation so that i think it might be possible to use a common implementation. One could delegat to the other or have I overlooked something?. 
Like this
```
    public static ImageLayout buildTileImageLayout(int dataBufferType, int imageWidth, int imageHeight, int level, Dimension defaultJAIReadTileSize) {
        return buildMosaicImageLayout(dataBufferType, imageWidth, imageHeight, level, defaultJAIReadTileSize,
                                      ImageUtils.computeLevelSize(defaultJAIReadTileSize.width, level),
                                      ImageUtils.computeLevelSize(defaultJAIReadTileSize.height, level));
    }
```
Is it important that the two parameters topLeftTileWidth and topLeftTileHeight are topleft? Isn't it more like maxTileWidth and maxTileHeight?

Also for the two buildMosaicMethods the first parameter should not be `Integer` but `int`.

Reply:
The TileIndexMultiLevelSource class invokes the 'buildMosaicImageLayout' method with null for dataBufferType, because the mosaic image for the index band of an S2 product is a constant color. Therefore the sample model is not created in the returned image layout object.
I created a new private method and moved the common source code of the two buildMosaicMethods.